src/clr-angular/button/button-loading/loading-button.ts

```diff
@@ -5,26 +5,91 @@
  */
 
 
-import {Component} from "@angular/core";
-import {LoadingListener} from "../../utils/loading/loading-listener";
-
+import {animate, keyframes, style, transition, trigger} from "@angular/animations";
+import {Component, ElementRef, EventEmitter, Output, Renderer2} from "@angular/core";
+import {ClrLoadingState, LoadingListener} from "../../utils/loading";
 
 @Component({
     selector: "button[clrLoading]",
     template: `
-        <span class="spinner spinner-inline" *ngIf="loading"></span>
-        <ng-content></ng-content>
+        <ng-container [ngSwitch]="state">
+            <span *ngSwitchCase="buttonState.LOADING">
+                <span @spinner class="spinner spinner-inline spinner-loading"></span>
+            </span>
+            <span *ngSwitchCase="buttonState.SUCCESS">
+                <span @validated class="spinner spinner-inline spinner-loading spinner-check"></span>
+            </span>
+            <span *ngSwitchCase="buttonState.DEFAULT" @defaultButton>
+                <ng-content></ng-content>
+            </span>
+        </ng-container>
     `,
-    providers: [{provide: LoadingListener, useExisting: ClrLoadingButton}]
+    providers: [{provide: LoadingListener, useExisting: ClrLoadingButton}],
+    animations: [
+        trigger("defaultButton",
+                [
+                    transition(":enter", [style({opacity: 0}), animate("200ms 100ms ease-in", style({opacity: 1}))]),
+                    // TODO: see if we can get leave animation to work before spinner's enter animation
+                    transition(":leave", [style({opacity: 0})])
+                ]),
+        trigger("spinner",
+                [
+                    transition(":enter", [style({opacity: 0}), animate("200ms 100ms ease-in", style({opacity: 1}))]),
+                    transition(":leave", [style({opacity: 1}), animate("100ms ease-out", style({opacity: 0}))])
+                ]),
+        trigger("validated",
```

Comment:
should we rename this state to `done` or `completed` instead of `validated`?

Reply:
Hmm.. I'm torn about that. Depending on the use case going from SPINNING to DEFAULT is considered "done" in some cases even though we skip the `validated` state.